sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobClient.java

```diff
@@ -381,7 +383,9 @@ public VoidResponse downloadWithResponse(OutputStream stream, BlobRange range, R
             .flatMapMany(res -> res.getValue()
                 .doOnNext(bf -> {
                     try {
-                        stream.write(bf.array());
+                        WritableByteChannel channel = Channels.newChannel(stream);
+                        channel.write(bf);
+                        channel.close();
```

Comment:
If the other comment is done this should be moved into the doOnCompletion or doOnTerminate